### PR TITLE
Microsoft graph files add support for large file

### PR DESCRIPTION
**Pull request by sapirshuker**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-4249

## Description
Allow uploading of large files.

## Minimum version of Cortex XSOAR
- [X] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

File: Tests/configure_and_test_integration_instances.py [hunk 1]
```diff
@@ -45,6 +45,7 @@
 NO_PROXY = ','.join([
     'oproxy.demisto.ninja',
     'oproxy-dev.demisto.ninja',
+    '*.sharepoint.com'
```
Comment (daryakoval) [inline Tests/configure_and_test_integration_instances.py:48]: What is this? Why it is needed?

Comment (ostolero) [reply Tests/configure_and_test_integration_instances.py:48]: We are experiencing SSL connection errors in the playbook, this is the fix suggested by @esharf

**Review by Shellyber — COMMENTED** [review]
Nice work

File: Packs/MicrosoftGraphFiles/TestPlaybooks/MS_Graph_Files_-_Test.yml [hunk 2]
```diff
@@ -1,22 +1,32 @@
 id: test_MsGraphFiles
 version: -1
+contentitemexportablefields:
```
Comment (Shellyber) [inline Packs/MicrosoftGraphFiles/TestPlaybooks/MS_Graph_Files_-_Test.yml:3]: Run the format commands on the test (those fields are unnecessary)

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_test.py [hunk 3]
```diff
@@ -306,3 +307,248 @@ def test_list_drives_in_site(command, args, response, expected_result, mocker):
...
+    upload_query_mock = mocker.patch.object(requests, 'put', side_effect=upload_response_side_effect)
+    upload_new_file_command(client, args)
+    assert validate_upload_attachments_flow(create_upload_mock, upload_query_mock)
+
+
+@pytest.mark.parametrize('client, args', UPLOAD_LARGE_FILE_COMMAND_ARGS)
+def test_upload_command_without_upload_session(mocker, client, args):
+    """
+        Given:
+            - An image to upload.
```
Comment (Shellyber) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_test.py:532]:
```suggestion
            - An image to upload (file size lower than 3)
```

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_test.py [hunk 4]
```diff
@@ -306,3 +307,248 @@ def test_list_drives_in_site(command, args, response, expected_result, mocker):
...
+        "ID": "some_id",
+    }
+}
+
+
+@pytest.mark.parametrize('client, args', UPLOAD_LARGE_FILE_COMMAND_ARGS)
+def test_upload_command_with_upload_session(mocker, client, args):
+    """
+        Given:
+            - An image to upload.
```
Comment (Shellyber) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_test.py:512]:
```suggestion
            - An image to upload with a size bigger than 3
```

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_test.py [hunk 5]
```diff
@@ -306,3 +307,248 @@ def test_list_drives_in_site(command, args, response, expected_result, mocker):
...
+            - return an result with upload session.
+     """
+    import requests
+    mocker.patch.object(demisto, 'getFilePath', return_value={'path': 'test_data/shark.jpg',
+                                                              'name': 'shark.jpg'})
+    create_upload_mock = mocker.patch.object(MsGraphClient, 'create_an_upload_session',
+                                             return_value=({"response": "", "uploadUrl": "test.com"}, "test.com"))
+    upload_query_mock = mocker.patch.object(requests, 'put', side_effect=upload_response_side_effect)
+    upload_new_file_command(client, args)
+    assert validate_upload_attachments_flow(create_upload_mock, upload_query_mock)
```
Comment (Shellyber) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_test.py:525]: Check here as well that the right function is called (with the session) and assert not for the regular function

**Review by Shellyber — COMMENTED** [review]
See my last notes

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_test.py (hunk 5 above)
Comment (sapirshuker) [reply Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_test.py:525]: there is a test in validate_upload_attachments_flow about checking that the upload session function called 
added a test for the function without the upload session

**Review by Shellyber — APPROVED** [review]

**Pull request merged**